scan.go

```diff
@@ -66,13 +66,21 @@ func (db *DB) scanIntoStruct(rows Rows, reflectValue reflect.Value, values []int
 	db.RowsAffected++
 	db.AddError(rows.Scan(values...))
 	joinedNestedSchemaMap := make(map[string]interface{})
+	fieldsWithValueMap := make(map[string]bool)
 	for idx, field := range fields {
 		if field == nil {
 			continue
 		}
 
 		if len(joinFields) == 0 || len(joinFields[idx]) == 0 {
-			db.AddError(field.Set(db.Statement.Context, reflectValue, values[idx]))
+			fieldIsEmbeddedPointerTypeStruct := len(field.BindNames) > 1 && len(field.StructField.Index) > 0 && field.StructField.Index[0] < 0
+			fieldValue := reflect.ValueOf(values[idx]).Elem()
+			if !fieldIsEmbeddedPointerTypeStruct && fieldValue.Kind() == reflect.Ptr && fieldValue.IsNil() {
+				db.AddError(field.Set(db.Statement.Context, reflectValue, field.DefaultValueInterface))
```

Comment:
Why use field.DefaultValueInterface? I don't think it is an expected behavior for most users.

Reply:
`field.DefaultValueInterface` ensures that I return the struct back with default/zero values for each field. i.e. nil for pointers and 0 for ints. It helps to reset the struct.

Reply:
@jinzhu any concerns with that?

Reply:
@jinzhu
I think it is expected usage based on your docs: https://gorm.io/docs/sql_builder.html#Scan-sql-Rows-into-struct
```golang
var user User
for rows.Next() {
  // ScanRows scan a row into user
  db.ScanRows(rows, &user)

  // do something
}
```

Reply:
@a631807682 @jinzhu ? any comments on the above? This is valid usage and should lower the memory allocations needed from my understanding

Reply:
@a631807682 @jinzhu ? any comments on the above? This is valid usage and should lower the memory allocations needed from my understanding